components/composed/contribution/CollectionContributionCreateDrawer/CollectionContributionCreateDrawer.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { graphql } from "react-relay";
+import { DialogProps } from "reakit/Dialog";
+import get from "lodash/get";
+import { useDrawerHelper, useRouteSlug } from "hooks";
+import type { CollectionContributionCreateDrawerQuery as Query } from "__generated__/CollectionContributionCreateDrawerQuery.graphql";
+import { Drawer } from "components/layout";
+import { QueryWrapper } from "components/api";
+import ContributionCreateForm from "../ContributionCreateForm";
+
+export default function CollectionContributionCreateDrawer({
+  dialog,
+  params,
+}: {
+  dialog: DialogProps;
+  params: Record<string, string>;
+}) {
+  const { t } = useTranslation();
+  const drawerHelper = useDrawerHelper();
+  const slug = useRouteSlug();
```

Comment:
Note: Need to check that we're on the contribution route, or any slug will get passed in to the drawer.

Reply:
I don't think this drawer (or any drawer should be using the route slug), because drawers should be completely separate from the context trigger them. That is, an "add contribution" drawer should work from anywhere, not just from this specific route. So, rather than getting the collection slug from the route, the button that triggers the drawer should include it in the drawerQuery so the drawer can pick it up, similar to what it's doing on edit views.